extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js

```diff
@@ -0,0 +1,190 @@
+        fill="#FF5252"
+      />
+    </svg>`;
+  }
+}
+
+const buildLogMessageTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`<div class="i-amphtml-story-dev-tools-log-message">
+    <div>
+      <span class="i-amphtml-story-dev-tools-log-type"></span>
+      <span> at </span>
+      <span class="i-amphtml-story-dev-tools-log-position"></span>
+    </div>
+    <span class="i-amphtml-story-dev-tools-log-description"></span>
+    <a class="i-amphtml-story-dev-tools-log-spec" target="_blank">Learn more</a>
+    <pre class="i-amphtml-story-dev-tools-log-code"></pre>
+  </div>`;
+};
+
+/** @const {string} */
+const TAG = 'AMP_STORY_DEV_TOOLS_LOGS';
+
+export class AmpStoryDevToolsTabLogs extends AMP.BaseElement {
+  /** @param {!Element} element */
+  constructor(element) {
+    super(element);
+
+    /** @private  {string} */
+    this.storyUrl_ = '';
+  }
+
+  /** @override */
+  buildCallback() {
+    this.storyUrl_ = this.element.getAttribute('data-story-url');
+    this.element.classList.add('i-amphtml-story-dev-tools-tab');
+  }
+
+  /** @override */
+  layoutCallback() {
+    return getInstance().then((validator) =>
+      this.validateUrl_(validator.sandbox.amp.validator, this.storyUrl_)
+    );
+  }
+
+  /**
+   * Validates a URL input, showing the errors on the screen.
+   *
+   * @private
+   * @param {!Validator} validator
+   * @param {string} url
+   * @return {!Promise}
+   */
+  validateUrl_(validator, url) {
+    return Services.xhrFor(this.win)
+      .fetchText(url)
+      .then((response) => {
+        userAssert(response.ok, 'Invalid story url');
+        return response.text();
+      })
+      .then((html) => {
+        const htmlLines = html.split('\n');
+        const validationResult = validator.validateString(html);
+        const errorList = validationResult.errors.map((error) => {
+          error.htmlLines = htmlLines.slice(error.line - 2, error.line + 3);
+          error.message = validator.renderErrorMessage(error);
+          return error;
+        });
+        return this.buildLogsList_(errorList);
+      })
+      .catch((error) => {
+        user().error(TAG, error);
+      });
+  }
+
+  /**
+   * @private
+   * @param {Array<Object>} errorList
+   * @return {!Promise}
+   */
+  buildLogsList_(errorList) {
+    const logsContainer = this.element.ownerDocument.createElement('div');
+    logsContainer.appendChild(this.buildLogsTitle_(errorList.length));
+    errorList.forEach((content) => {
+      const logEl = buildLogMessageTemplate(this.element);
+      logEl.querySelector('.i-amphtml-story-dev-tools-log-type').textContent =
+        content.code;
+      const codeEl = logEl.querySelector('.i-amphtml-story-dev-tools-log-code');
+      content.htmlLines.forEach((l, i) => {
+        const lineEl = this.element.ownerDocument.createElement('div');
```

Comment:
a11y: These should be text elements. 
Maybe they could be `spans` with `display: block`.